lmfit/minimizer.py

```diff
@@ -499,6 +507,124 @@ def scalar_minimize(self, method='Nelder-Mead', params=None, **kws):
 
         return result
 
+    def emcee(self, params=None, steps=1000, nwalkers=100, burn=0, thin=1,
+              ntemps=1):
+        """
+        Bayesian sampling of the posterior distribution for the parameters
+        using the `emcee` Markov Chain Monte Carlo package.
+        The method assumes that the prior is Uniform.
+        You need to have `emcee` and `pandas` installed to use this method.
+
+        Parameters
+        ----------
+        steps : int, optional
+            How many samples you would like to draw from the posterior
+            distribution?
+        burn : int, optional
+            Discard this many samples from the start of the sampling regime.
+        thin : int, optional
+            Only accept 1 in every `thin` samples.
+        ntemps : int, optional
+            If `ntemps > 1` perform a Parallel Tempering.
+
+        Returns
+        -------
+        result, chain : MinimizerResult, pandas.DataFrame
+            MinimizerResult object contains updated params, fit statistics, etc.
+            The `chain` contains the samples. Has shape (steps, ndims). Access
+            chain values for a particular parameter using chain[parname].
+        """
+        if not HAS_EMCEE:
+            raise NotImplementedError('You must have emcee to use the emcee '
+                                      'method')
+
+        result = self.prepare_fit(params=params)
+        vars   = result.init_vals
+        params = result.params
+
+        # Removing internal parameter scaling. We could possibly keep it,
+        # but I don't know how this affects the emcee sampling.
+        bounds_varying = []
+        for i, par in enumerate(params):
+            param = params[par]
+            vars[i] = param.from_internal(param.value)
+            param.from_internal = lambda val: val
+            lb, ub = param.min, param.max
+            if lb is None or lb is np.nan:
+                lb = -np.inf
+            if ub is None or ub is np.nan:
+                ub = np.inf
+            bounds_varying.append((lb, ub))
+
+        bounds_varying = np.array(bounds_varying)
+
+        self.nvarys = len(result.var_names)
+
+        def lnlike(theta):
+            # log likelihood
+            return -0.5 * self.penalty(theta)
```

Comment:
@Tillsten, so that I understand more fully, and with reference to the emcee equation for the log likelihood function, what terms do you think we're ignoring?